src/hotspot/share/opto/vtransform.cpp

```diff
@@ -158,11 +158,9 @@ void VTransform::apply_speculative_alignment_runtime_checks() {
 
     const GrowableArray<VTransformNode*>& vtnodes = _graph.vtnodes();
     for (int i = 0; i < vtnodes.length(); i++) {
-      VTransformVectorNode* vtn = vtnodes.at(i)->isa_Vector();
+      VTransformMemVectorNode* vtn = vtnodes.at(i)->isa_MemVector();
       if (vtn == nullptr) { continue; }
-      MemNode* p0 = vtn->nodes().at(0)->isa_Mem();
-      if (p0 == nullptr) { continue; }
-      const VPointer& vp = vpointer(p0);
+      const VPointer& vp = vtn->vpointer();
```

Comment:
We can check for `MemVector` directly, and then we know that they all represent `Mem` nodes and they all have a `vpointer`.